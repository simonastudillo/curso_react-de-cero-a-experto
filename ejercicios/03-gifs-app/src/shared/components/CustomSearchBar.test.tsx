import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import { CustomSearchBar } from "./CustomSearchBar";

describe('CustomSearchBar', () => {

   test('Should render CustomSearchBar correctly', () => {
      const { container } = render(<CustomSearchBar onQuery={() => { }} />);

      const input = screen.getByRole('textbox');
      const button = screen.getByRole('button');

      expect(container).toMatchSnapshot();
      expect(input).toBeDefined();
      expect(button).toBeDefined();
   });

   test('Should call onQuery with the current value after 700ms', async () => {
      const onQuery = vi.fn();
      const inputValue = 'test';
      const { container } = render(<CustomSearchBar onQuery={onQuery} />);

      const input = screen.getByRole('textbox');
      fireEvent.change(input, { target: { value: `${inputValue}` } });

      // await new Promise(resolve => setTimeout(resolve, 701));
      await waitFor(() => {
         expect(onQuery).toHaveBeenCalled();
         expect(onQuery).toHaveBeenCalledWith(inputValue);
      });
   });

});