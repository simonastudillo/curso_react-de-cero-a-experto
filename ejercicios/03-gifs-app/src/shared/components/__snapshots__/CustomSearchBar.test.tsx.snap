// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomSearchBar > Should render CustomSearchBar correctly 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      placeholder="Buscar"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
</div>
`;
